static/app/components/sidebar/help.tsx

```diff
@@ -29,15 +31,21 @@ type Props = Pick<
 
 function SidebarHelp({orientation, collapsed, hidePanel, organization}: Props) {
   const user = useUser();
-  const {shouldShowHelpMenuDot, onOpenHelpMenu} = useNavPrompts({
+  const navPrompts = useNavPrompts({
     collapsed,
     organization,
   });
+  const chonkPrompts = useChonkPrompt();
   const {mutate: mutateUserOptions} = useMutateUserOptions();
   const openForm = useFeedbackForm();
 
+  const onHelpMenuOpen = useCallback(() => {
+    navPrompts.onOpenHelpMenu();
+    chonkPrompts.dismissDotIndicatorPrompt();
+  }, [navPrompts, chonkPrompts]);
```

Comment:
there’s no need to memoize this with `useCallback`. For one, `navPrompts.onOpenHelpMenu` is an inline function upon creation:

https://github.com/getsentry/sentry/blob/d5f5d49ec068ac8503e71508596736580c901ef2/static/app/views/nav/useNavPrompts.tsx#L42-L46

and also, we’re only using the resulting function to pass it to `onOpen` of `DeprecatedDropdownMenu` which is neither a pure component nor does it use the function in memoization or effects (which it shouldn’t do anyway).

That means the memoization will be re-calculated on every render anyways, and we don’t need it. I’m wondering where the instinct to wrap things in `useMemo` and `useCallback` in the sentry code-base comes from 🤔. This doesn’t make the code any faster, if anything, it’ll make it “slower” because there will be one additional caching react has to perform. It also makes the codebase a lot harder to read if this is done in every component.

```suggestion
  const onHelpMenuOpen = () => {
    navPrompts.onOpenHelpMenu();
    chonkPrompts.dismissDotIndicatorPrompt();
  };
```

Reply:
The instinct here is self-preservational. The codebase is full of effects under untested paths, and I've had changes like these blow up on me so often that I err on the safe side of things. That said though, this is unnecessary in this situation indeed, and should be safe to remove

Reply:
understandable, we need to work on the root cause :)